include/roboteam_ai/utilities/Dealer.h

```diff
@@ -69,14 +70,14 @@ class Dealer {
     virtual ~Dealer() = default;  // needed for test
 
     /**
-     * Distributes the role between the friendly robots considering all information given in the flagMap
-     * !!! allRobots and flagMap are copies as they are modified to reduce computations after the forcedID's.
-     * @param allRobots vector containing all friendly robots
-     * @param flagMap information map with all information for each role (with key -> roleName)
+     * Distributes the role between the friendly robots considering all information given in the role_map
+     * !!! robots and role_map are copies as they are modified to reduce computations after the forcedID's.
+     * @param robots vector containing all friendly robots
+     * @param role_map information map with all information for each role (with key -> roleName)
      * @param stpInfoMap
      * @return a vector with the roleName and the Robot that should get the role
      */
-    std::unordered_map<std::string, v::RobotView> distribute(std::vector<v::RobotView> allRobots, FlagMap flagMap, const std::unordered_map<std::string, stp::StpInfo> &stpInfoMap);
+    std::unordered_map<std::string, v::RobotView> distribute(std::vector<v::RobotView> robots, FlagMap role_map, const std::unordered_map<std::string, stp::StpInfo> &stpInfoMap);
```

Comment:
Keep in mind that a robot view is still a reference to the original robot

Reply:
I'm not sure what you're saying. Do you mean the segfault danger is still there when the original robots disappear?

Reply:
Yeah exactly, std::vector<robot> would be better, you can always construct a view from that